### PR TITLE
add jepa

### LINKED ISSUE #4512: Add model: v-jepa2
### Model link on Hugging Face

https://huggingface.co/collections/facebook/v-jepa-2

### Arxiv link

_No response_

### Description of the model

_No response_

### PR DESCRIPTION
**Pull request by Samoed**
Close https://github.com/embeddings-benchmark/mteb/issues/4512

- [X] I have filled out the ModelMeta object to the extent possible
- [X] I have ensured that my model can be loaded using
  - [X] `mteb.get_model(model_name, revision)` and
  - [X] `mteb.get_model_meta(model_name, revision)`
- [X] I have tested the implementation works on a representative set of tasks.
- [X] The model is public, i.e., is available either as an API or the weights are publicly available to download
- [ ] I reproduced results from the original paper (if applicable) on at least one benchmark, and I am including the results in the PR description.

```json
{
  "dataset_revision": "59a874899eb241993794a3454c37829727c3b559",
  "task_name": "BreakfastClassification",
  "mteb_version": "2.12.25",
  "scores": {
    "test": [
      {
        "scores_per_experiment": [
          {
            "accuracy": 0.114943,
            "f1": 0.117577,
            "f1_weighted": 0.112162,
            "precision": 0.115999,
            "precision_weighted": 0.117049,
            "recall": 0.128077,
            "recall_weighted": 0.114943,
            "ap": null,
            "ap_weighted": null
          },
          {
            "accuracy": 0.103448,
            "f1": 0.10537,
            "f1_weighted": 0.101859,
            "precision": 0.107197,
            "precision_weighted": 0.107933,
            "recall": 0.113571,
            "recall_weighted": 0.103448,
            "ap": null,
            "ap_weighted": null
          },
          {
            "accuracy": 0.103448,
            "f1": 0.095497,
            "f1_weighted": 0.097577,
            "precision": 0.097885,
            "precision_weighted": 0.101547,
            "recall": 0.1025,
            "recall_weighted": 0.103448,
            "ap": null,
            "ap_weighted": null
          },
          {
            "accuracy": 0.081395,
            "f1": 0.075447,
            "f1_weighted": 0.079767,
            "precision": 0.07671,
            "precision_weighted": 0.087058,
            "recall": 0.091111,
            "recall_weighted": 0.081395,
            "ap": null,
            "ap_weighted": null
          },
          {
            "accuracy": 0.127907,
            "f1": 0.117102,
            "f1_weighted": 0.126733,
            "precision": 0.126737,
            "precision_weighted": 0.140638,
            "recall": 0.11974,
            "recall_weighted": 0.127907,
            "ap": null,
            "ap_weighted": null
          }
        ],
        "accuracy": 0.106228,
        "f1": 0.102198,
        "f1_weighted": 0.103619,
        "precision": 0.104906,
        "precision_weighted": 0.110845,
        "recall": 0.111,
        "recall_weighted": 0.106228,
        "ap": NaN,
        "ap_weighted": NaN,
        "main_score": 0.106228,
        "hf_subset": "default",
        "languages": [
          "eng-Latn"
        ]
      }
    ]
  },
  "evaluation_time": 601.7810099124908,
  "kg_co2_emissions": null,
  "date": 1777490655.945098
}
```

### CODE REVIEW

File: mteb/models/model_implementations/facebook_jepa.py [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+        hf_split: str,
+        hf_subset: str,
+        prompt_type: PromptType | None = None,
+        show_progress_bar: bool = True,
+        **kwargs: Unpack[EncodeKwargs],
+    ) -> Array:
+        inputs.collate_fn = FramesCollator(
+            fps=self.fps,
+            max_frames=10,
+            num_frames=self.num_frames,
```
Comment (AdnanElAssadi) [inline mteb/models/model_implementations/facebook_jepa.py]:
```suggestion
            max_frames=self.max_frames,
            num_frames=self.num_frames,
```

File: mteb/models/model_implementations/facebook_jepa.py [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+            videos = batch["video"]
+            max_frames = max(v.shape[0] for v in videos)
+            padded = [
+                torch.cat(
+                    [v, v[-1:].expand(max_frames - v.shape[0], *v.shape[1:])], dim=0
+                )
+                if v.shape[0] < max_frames
+                else v
+                for v in videos
+            ]
```
Comment (AdnanElAssadi) [inline mteb/models/model_implementations/facebook_jepa.py:84]: is this what other processors do internally for padding?

File: mteb/models/model_implementations/facebook_jepa.py (hunk 1 above)
Comment (Samoed) [reply mteb/models/model_implementations/facebook_jepa.py]: Oh, yes. Was testing, because this is not possible to run this model on mps

File: mteb/models/model_implementations/facebook_jepa.py (hunk 2 above)
Comment (Samoed) [reply mteb/models/model_implementations/facebook_jepa.py:84]: I'm not sure. This is not possible to pass `padding=True` to `AutoVideoProcessor`. They combine videos and pass them after in qwen3 modelling https://github.com/huggingface/transformers/blob/a8f43eca15b8d1c63deb33f6b97dfab30419e5da/src/transformers/models/qwen3_vl/video_processing_qwen3_vl.py#L257-L264

**Review by AdnanElAssadi56 — APPROVED** [review]

**Comment by isaac-chung** [comment]
Good to merge?

**Comment by Samoed** [comment]
Not for now, I'll run this model on some tasks to see if results are matching

**Comment by Samoed** [comment]
I tried to run, but they report only results on classification tasks with different crops `16 frames clip, 2 temporal crops, 3 spatial crops`, so I think we can't reproduce this directly
